cms/djangoapps/contentstore/outlines.py

```diff
@@ -306,7 +327,18 @@ def get_outline_from_modulestore(course_key) -> Tuple[CourseOutlineData, List[Co
         # Pull course with depth=3 so we prefetch Section -> Sequence -> Unit
         course = store.get_course(course_key, depth=3)
         sections_data = []
-        for section in course.get_children():
+        sections = course.get_children()
+        for i, section in enumerate(sections):
+            for j in range(i + 1, len(sections)):
+                if i == j:
+                    break
+                section_1_children = section.children
+                section_2_children = sections[j].children
+                for index, child in enumerate(section_2_children):
+                    if child in section_1_children:
+                        acyclical_errors = _error_for_duplicate_child(sections[j], sections[j].get_children()[index])
+                        content_errors.append(acyclical_errors)
```

Comment:
I think this part of the code is more complex than it needs to be. Maybe try using a dictionary to hold the mapping of sequence usage keys to the first section they were in, instead of that innermost for loop? So maybe something like this:

```python
sequences_to_first_section = {}
for section in course.get_children():
    for sequence in section.get_children():
        if sequence.location in sequences_to_first_section:
            content_errors.append(_error_for_duplicate_child(section, sequence))
        else:
            sequences_to_first_section[sequence.location] = section.location
```

(Caveat: I have not tested that at all.)

Reply:
I also thought it was a bit clunky but wasn't sure how to simplify it. I will check out your suggestion.